universal-ctags: bump to p5.9.20210124.0 from git tags

bobrik opened the pull request:
###### Motivation for this change

Switching to python2 to python3 seems like a reasonable thing to do.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bobrik committed:
universal-ctags: bump to p5.9.20210124.0 from git tags

bobrik committed:
universal-ctags: switch from python2 to python3

bobrik commented on pkgs/development/tools/misc/universal-ctags/default.nix:
```diff
@@ -1,17 +1,17 @@
+stdenv.mkDerivation rec {
   pname = "universal-ctags";
-  version = "unstable-2019-07-30";
+  version = "p5.9.20210124.0";
```
`p` here is a bit confusing, since I cannot do uninstall of `universal-ctags`. Not sure what's the right way to address this.

SuperSandro2000 replied:
The version number is probably just `5.9.20210124.0` and the p can be added when fetching.

bobrik replied:
That was my guess as well, updated.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111216` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>global</li>
<li>universal-ctags</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>global</li>
<li>universal-ctags</li>
</ul>
</details>

bobrik commented on pkgs/development/tools/misc/universal-ctags/default.nix:
```diff
@@ -1,17 +1,17 @@
     repo = "ctags";
-    rev = "920e7910146915e5cae367bc9f135ffd8b042042";
+    rev = version;
     sha256 = "14n3ix77rkhq6vq6kspmgjrmm0kg0f8cxikyqdq281sbnfq8bajn";
```
It doesn't seem ok that I changed the version without changing sha256 sum and everything worked.

bobrik replied:
I looked at https://discourse.nixos.org/t/troubleshooting-fetch-commands-like-fetchfromgithub/6159 and it's expected.

tomberek commented:
Result of `nixpkgs-review pr 111216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>global</li>
    <li>universal-ctags</li>
  </ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111216` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>global</li>
<li>universal-ctags</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>global: log was empty</li>
<li>universal-ctags: log <a href="https://termbin.com/8gyl">https://termbin.com/8gyl</a></li>
  </ul>
</details>

bobrik committed:
universal-ctags: downgrade to p5.9.20201206.0

Newer versions fail on aarch64-linux:

```
  checking for aarch64-unknown-linux-gnu-gcc... aarch64-unknown-linux-gnu-gcc
  checking whether the compiler supports GNU C... yes
  checking whether aarch64-unknown-linux-gnu-gcc accepts -g... yes
  checking for aarch64-unknown-linux-gnu-gcc option to enable C11 features... (cached) none needed
  checking whether aarch64-unknown-linux-gnu-gcc understands -c and -o together... yes
  checking dependency style of aarch64-unknown-linux-gnu-gcc... none
  checking whether the C compiler works... no
  configure: error: in `/build/source':
  configure: error: C compiler cannot create executables
  See `config.log' for more details
```

The following PR is likely culprit:

* https://github.com/universal-ctags/ctags/pull/2747

bobrik commented:
Pushed another commit to downgrade the version to make `aarch64-linux` build. The issue is very similar to this one:

* https://github.com/alsa-project/alsa-firmware/issues/3

The common thing being `ax_prog_cc_for_build.m4` and the error log message. I don't know how to fix it properly.

cc @flokli

bobrik commented:
Huh, the tests fail in ofborg:

```
Testing lregex-kind-letter
------------------------------------------------------------
stdout                                                      failed (diff: /build/source/Tmain/lregex-kind-letter.d/stdout-diff.txt)
stderr                                                      failed (diff: /build/source/Tmain/lregex-kind-letter.d/stderr-diff.txt)
..
Testing lregex-kind-name
------------------------------------------------------------
stdout                                                      failed (diff: /build/source/Tmain/lregex-kind-name.d/stdout-diff.txt)
stderr                                                      failed (diff: /build/source/Tmain/lregex-kind-name.d/stderr-diff.txt)
```

But pass for me locally (on both aarch64-linux and aarch64-darwin):

```
Testing lregex-kind-letter
------------------------------------------------------------
stdout                                                      passed
stderr                                                      passed
...
Testing lregex-kind-name
------------------------------------------------------------
stdout                                                      passed
stderr                                                      passed
```

How can I debug this?

bobrik commented:
Found why it was working for me locally: sandboxing was off. The reason why tests were failing with sandboxing enabled was usage of hardcoded `/bin/echo` in tests:

* https://github.com/universal-ctags/ctags/blob/p5.9.20201206.0/Tmain/utils.sh#L125-L131

@SuperSandro2000, builds are green now, this is ready to go now.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>global</li>
<li>universal-ctags</li>
</ul>
</details>
